spec/acceptance/namespaced_factories_spec.rb

```diff
@@ -0,0 +1,87 @@
+describe "Namespaced factories" do
+  before do
+    define_model("Category", name: :string)
+    define_model("Movies::Category", name: :string)
+  end
+
+  context "by default" do
+    before do
+      FactoryBot.define do
+        factory :category do
+          name { "Classics" }
+        end
+
+        with_namespace(:Movies) do
+          factory :category do
+            name { "Emmy Winners" }
+          end
+        end
+      end
+    end
+
+    it "creates a factory prefixed with the namespace" do
+      instance = FactoryBot.build(:movies_category)
+
+      expect(instance).to be_kind_of(Movies::Category)
+      expect(instance.name).to eql("Emmy Winners")
+    end
+
+    it "does not collide with un-namespaced factories" do
+      instance = FactoryBot.build(:category)
+
+      expect(instance).to be_kind_of(Category)
+      expect(instance.name).to eql("Classics")
+    end
+  end
+
+  context "when prefixes are not required" do
+    it "uses the namespaced factory without the prefix" do
+      FactoryBot.define do
+        with_namespace(:Movies, require_prefix: false) do
```

Comment:
I think I could go either way, with a slight preference to not include a default so that we don't introduce breaking changes by swapping the default later.